word_racer/session.py

```diff
@@ -0,0 +1,295 @@
+                await m[0].add_reaction(m[1])
+            else:
+                await asyncio.sleep(.25)
+
+    async def finish_round(self):
+        await self.send_round_table(f"- Round {self.level+1} over! \n Round scores:")
+        if self.level in [1,2]:
+            await self.send_table("Total scores so far:")
+        
+        table = ""
+        for word, score in self.valid_words.most_common():
+            claim = "none"
+            if word in self.claims:
+                claim = self.claims[word]
+            table += f"{word}\t{score}\t{claim}\n"
+            if len(table) > 1900:
+                await self.ctx.send(box(table, lang="diff"))
+                table = ""
+        if table != "":
+            await self.ctx.send(box(table, lang="diff"))
+    
+    async def timer_task(self):
+        reveal = 3
+        while reveal:
+            await asyncio.sleep(30)
+            msg = f"{30*reveal} seconds remaining in round {self.level+1}. {len(self.valid_words)-len(self.claims)} words left to find."
+            await self.send_round_table(msg)
+            reveal -= 1
+        await asyncio.sleep(30)
+
+    def check_message(self, message: discord.Message):
+        early_exit = message.channel != self.ctx.channel or message.author == self.ctx.guild.me
+        if early_exit:
+            return
+        guess = message.content.lower()
+        if not re.match(r"[a-z]{3,}", guess): # check if guess is a string of letters
+            return
+        if guess not in self.valid_words:
+            # Wrong answer handling
+            self.reaction_queue.append((message, "\N{CROSS MARK}"))
+            self.scores[message.author] -= 1
+            self.round_scores[message.author] -= 1
+            return
+        if guess in self.claims:
+            # Slow answer handling
+            self.reaction_queue.append((message, "\U0001f501"))
+            return
+        # Correct answer handling
+        self.claims[guess] = message.author
+        self.scores[message.author] += self.valid_words[guess]
+        self.round_scores[message.author] += self.valid_words[guess]
+        self.reaction_queue.append((message, "\N{WHITE HEAVY CHECK MARK}"))
+        return 
+        
+    async def end_game(self):
+        """End the game and display scores."""
+        if self.scores:
+            await self.send_table("Final results:")
+        self.stop()
+
+    async def send_table(self, msg):
+        """Send a table of scores to the session's channel."""
+        table = f"+ {msg} \n\n"
+        for user, score in self.scores.most_common():
+            table += f"+ {user}\t{score}\n"
+        await self.ctx.send(box(table, lang="diff"))
+    
+    async def send_round_table(self, msg):
+        """Send a table of round scores to the session's channel."""
+        table = f"+ {msg} \n\n"
+        for user, score in self.round_scores.most_common():
+            table += f"+ {user}\t{score}\n"
```

Comment:
cf. message about tab interaction with high string length variance, this time for user names